### PR TITLE
Bringing BM Stuff to MMD Lib

### PR DESCRIPTION
**Pull request by faceofcat**

### CODE REVIEW

File: src/main/java/com/mcmoddev/lib/data/MaterialNames.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
+package com.mcmoddev.lib.data;
```
Comment (jriwanek) [inline src/main/java/com/mcmoddev/lib/data/MaterialNames.java:1]: This is part of BMe, not MMDlib, if it's going to exist at all in the lib, it should only contain names of materials in vanilla

**Review by jriwanek — CHANGES_REQUESTED** [review]
also, lib.crafting and lib.energy probably shouldn't be removed unless I'm missing something

File: src/main/java/com/mcmoddev/lib/data/MaterialNames.java (hunk 1 above)
Comment (faceofcat) [reply src/main/java/com/mcmoddev/lib/data/MaterialNames.java:1]: I agree...

but... I moved this here to get rid of 20+ compile errors for tests against material names.

I say, let's get this to work and make an issue so we don't forget about refactoring this part.

Comment (jriwanek) [reply src/main/java/com/mcmoddev/lib/data/MaterialNames.java:1]: kk

**Review by jriwanek — APPROVED** [review]

File: src/main/java/com/mcmoddev/lib/block/BlockMMDTrapDoor.java [hunk 2]
```diff
@@ -23,29 +24,43 @@
...
-		world.playEvent(player, ((Boolean) newState.getValue(BlockTrapDoor.OPEN)) ? 1003 : 1006, coord, 0);
-		return true; */
+		}
+	}
+
+	// Magic Number mappings from net.minecraft.client.renderer.RenderGlobal#playEvent
+	// SoundEvents.BLOCK_IRON_TRAPDOOR_OPEN = 1037
+	// SoundEvents.BLOCK_WOODEN_TRAPDOOR_OPEN = 1007
+	// SoundEvents.BLOCK_IRON_TRAPDOOR_CLOSE = 1036
+	// SoundEvents.BLOCK_WOODEN_TRAPDOOR_CLOSE = 1013
```
Comment (dshadowwolf) [inline src/main/java/com/mcmoddev/lib/block/BlockMMDTrapDoor.java:53]: Can we use the constants instead of the magic numbers ?

**Review by dshadowwolf — APPROVED** [review]
Not seeing anything really objectionable, though we might want to have lib.crafting remain in some form - perhaps merged off to a shared namespace with the crusher recipe and repair recipe stuff.

And... lib.energy could probably remain as well - though I'm not sure if lib.energy is the right namespace, it works for now.

File: src/main/java/com/mcmoddev/lib/data/MaterialNames.java (hunk 1 above)
Comment (faceofcat) [reply src/main/java/com/mcmoddev/lib/data/MaterialNames.java:1]: all non-vanilla materials removed.

Comment (jriwanek) [reply src/main/java/com/mcmoddev/lib/data/MaterialNames.java:1]: Including several vanilla materials being removed too:
diamond, emerald, gold, obsidian, prismarine, redstone, stone

File: src/main/java/com/mcmoddev/lib/init/Fluids.java [hunk 3]
```diff
@@ -92,15 +90,16 @@ protected static BlockFluidClassic addFluidBlock(@Nonnull final MMDMaterial mate
 			return null;
 		}
 
-		if (!name.equals(MaterialNames.MERCURY)) {
-			block = new BlockFluidClassic(material.getFluid(), Material.LAVA);
-		} else {
+		// TODO: make it into a flag/stat or something
+//		if (!name.equals(MaterialNames.MERCURY)) {
+//			block = new BlockFluidClassic(material.getFluid(), Material.LAVA);
```
Comment (jriwanek) [inline src/main/java/com/mcmoddev/lib/init/Fluids.java]: line 95 is the generic bit, the rest which you've inadvertently force enabled is mercury specific

**Review by jriwanek — CHANGES_REQUESTED** [review]

**Review by jriwanek — APPROVED** [review]

**Pull request merged**